### PR TITLE
Fix birthday determination bug in `user#age_in_years`

### PR DESCRIPTION
**Pull request by dunkOnIT**
We had a bug in the forum age restriction where a user born in the same year, earlier month and later date caused the system to think their birthday had not yet passed - fixed in this PR, and added a test for it.

### CODE REVIEW

**Comment by dunkOnIT** [comment]
No good reason - that's much simpler!

File: app/models/user.rb [hunk 1]
```diff
@@ -492,14 +492,8 @@ def senior_delegate_for?(user)
...
 
-  def age_in_years
-    years_since_birth = Date.today.year - dob.year
-    birthday_happened = Date.today.day >= dob.day && Date.today.month >= dob.month
-    birthday_happened ? years_since_birth : years_since_birth - 1
-  end
-
   def below_forum_age_requirement?
-    age_in_years < FORUM_AGE_REQUIREMENT
+    (Date.today - 13.years) < dob
```
Comment (gregorbg) [inline app/models/user.rb]: Perhaps this should be `<=` so that you can access the forums on the day of your 13th birthday

Comment (gregorbg) [reply app/models/user.rb]: Also, it seems you're completely ditching `FORUM_AGE_REQUIREMENT` now, is there a special reason for this? `Date.today - FORUM_AGE_REQUIREMENT.years` should :tm: work

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -761,29 +761,6 @@
...
-      user.dob = Date.today.advance(years: -10)
-      expect(user.age_in_years).to be(10)
-    end
-
-    it 'doesnt count the year if before user birthday' do
-      user.dob = Date.today.advance(years: -10, days: 1)
-      expect(user.age_in_years).to be(9)
-    end
-  end
-
```
Comment (gregorbg) [inline spec/models/user_spec.rb:786]: These very relevant test cases should be migrated to the `below_forum_age_requirement` part below -- unless they already are?
I mean, fake a user with these "edge case" birthdates and then run the `below?` method on them.

**Review by gregorbg — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 1 above)
Comment (dunkOnIT) [reply app/models/user.rb]: > Perhaps this should be <= so that you can access the forums on the day of your 13th birthday

That case is already covered in the current logic - see `user_spec#800`

And changed to use the constant :+1:

File: spec/models/user_spec.rb (hunk 2 above)
Comment (dunkOnIT) [reply spec/models/user_spec.rb:786]: We already have tests covering `below_forum_age_requirement` - some cases were extra here because of the implementation quirks of `age_in_years`
